src/sacn/mem.c

```diff
@@ -525,29 +527,60 @@ bool add_lost_source(SourcesLostNotification* sources_lost, const EtcPalUuid* ci
 {
   SACN_ASSERT(sources_lost);
   SACN_ASSERT(cid);
+  SACN_ASSERT(name);
 
   CHECK_ROOM_FOR_ONE_MORE(sources_lost, lost_sources, SacnLostSource, SACN_RECEIVER_MAX_SOURCES_PER_UNIVERSE);
 
   sources_lost->lost_sources[sources_lost->num_lost_sources].cid = *cid;
   ETCPAL_MSVC_NO_DEP_WRN strcpy(sources_lost->lost_sources[sources_lost->num_lost_sources].name, name);
   sources_lost->lost_sources[sources_lost->num_lost_sources].terminated = terminated;
   ++sources_lost->num_lost_sources;
+
   return true;
 }
 
-//CHRISTIAN TODO:
 /*
  * Add a new found source to a SourcesFoundNotification.
  *
  * [out] sources_found SourcesFoundNotification instance to which to append the found source.
- * ????????????
+ * [in] cid CID of the source that was found.
+ * [in] name Name of the source that was found; copied into the notification.
+ * [in] from_addr The address from which we received these initial packets.
+ * [in] priority The source's per-universe priority.
+ * [in] values The DMX (startcode 0) data. The library owns this data, and the memory is only guaranteed to be valid for the
+ * length of the SacnSourcesFound callback.
+ * [in] values_len The count of valid values.
+ * [in] preview Whether or not we only saw startcode 0 packets with the preview flag set.
+ * [in] per_address The per-address priority (startcode 0xdd) data, if the source is sending it. The library owns this data, and the
+ * memory is only guaranteed to be valid for the length of the SacnSourcesFound callback.
+ * [in] per_address_len The count of valid priorities.
  * Returns true if the source was successfully added, false if memory could not be allocated.
  */
-bool add_found_source(SourcesFoundNotification* sources_found /*,  ??????? */)
+bool add_found_source(SourcesFoundNotification* sources_found, const EtcPalUuid* cid, const char* name,
```

Comment:
Can you use the SacnHeaderData struct to reduce the number of parameters to this function?